VMR build failure for nuget-client repo

```
  /vmr/.dotnet/sdk/8.0.100-preview.4.23260.1/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1275,3): error MSB4019: The imported project "/vmr/.dotnet/sdk/8.0.100-preview.4.23260.1/Sdks/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.targets" was not found. Confirm that the expression in the Import declaration ";/vmr/.dotnet/sdk/8.0.100-preview.4.23260.1/Sdks/Microsoft.NET.Sdk/targets/../../Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.targets" is correct, and that the file exists on disk. [/vmr/src/nuget-client/src/NuGet.Clients/NuGet.Console/NuGet.Console.csproj]
```

This was introduced by something within the range of these commits: https://github.com/dotnet/dotnet/commit/85b801612ad392328eae4ec322bdccc85ada8e31 and https://github.com/dotnet/dotnet/commit/aecc50daa66f0db0f162a2138ac161e47b23109f.

This occurs in the CentOSStream8_Online_PreviousSourceBuiltSdk_x64 build leg.

--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -1,3 +1,4 @@
 <Project>
   <Import Project="build\common.targets" Condition="'$(IsCrossTargetingBuild)' == 'true' And '$(_WasCommonPropsImported)' != 'true'" />
+  <Import Project="eng\source-build\ExcludeFromSourceBuild.AfterCommonTargets.targets" Condition="'$(DotNetBuildFromSource)' == 'true'" />
 </Project>

--- a/eng/source-build/ExcludeFromSourceBuild.AfterCommonTargets.targets
+++ b/eng/source-build/ExcludeFromSourceBuild.AfterCommonTargets.targets
@@ -1,0 +1,21 @@
+<Project>
+  <!--
+    If a project specifies ExcludeFromSourceBuild=true during a source build suppress all targets and emulate a no-op
+    (empty common targets like Restore, Build, Pack, etc.).
+    Inspired by https://github.com/dotnet/arcade/blob/d9275ca3f692b5f97702bb6717a3bf78afedbf38/src/Microsoft.DotNet.Arcade.Sdk/tools/ExcludeFromBuild.BeforeCommonTargets.targets
+  -->
+  <PropertyGroup>
+    <_SuppressAllTargets>false</_SuppressAllTargets>
+    <_SuppressAllTargets Condition="'$(DotNetBuildFromSource)' == 'true' and '$(ExcludeFromSourceBuild)' == 'true'">true</_SuppressAllTargets>
+    <!-- If excluding, then disable a restore warning, which will fire on newer SDKs, as well as set the NuGetRestoreTargets property to empty,
+         which will avoid importing the restore targets inside the .NET SDK. If the restore targets exist, then static graph restore will attempt tpo
+         execute. -->
+    <DisableWarnForInvalidRestoreProjects Condition="'$(_SuppressAllTargets)' == 'true'">true</DisableWarnForInvalidRestoreProjects>
+    <NuGetRestoreTargets Condition="'$(_SuppressAllTargets)' == 'true'">$(MSBuildThisFileDirectory)NoRestore.targets</NuGetRestoreTargets>
+    <!-- When a project is using the .NET SDK, but with the "UseWpf" property, there will be an attempt to import the windows desktop SDK targets.
+         These are not available in certain circumstances, like linux source build. -->
+    <ImportWindowsDesktopTargets Condition="'$(_SuppressAllTargets)' == 'true'">false</ImportWindowsDesktopTargets>
+  </PropertyGroup>
+
+  <Import Project="Noop.proj" Condition="'$(_SuppressAllTargets)' == 'true'" />
+</Project>

--- a/eng/source-build/Noop.proj
+++ b/eng/source-build/Noop.proj
@@ -1,0 +1,6 @@
+<Project>
+  <Target Name="Restore" />
+  <Target Name="Build" />
+  <Target Name="Pack" />
+  <Target Name="Test" />
+</Project>

--- a/src/NuGet.Clients/NuGet.Console/NuGet.Console.csproj
+++ b/src/NuGet.Clients/NuGet.Console/NuGet.Console.csproj
@@ -8,6 +8,7 @@
     <Description>Package Manager PowerShell Console implementation.</Description>
     <TargetFramework>$(NETFXTargetFramework)</TargetFramework>
     <UseWpf>true</UseWpf>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj
+++ b/src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj
@@ -11,6 +11,7 @@
     <UseWinForms>true</UseWinForms>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
     <TargetFramework>$(NETFXTargetFramework)</TargetFramework>
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
   </PropertyGroup>
 
   <ItemGroup>

